crates/symbolicator/src/services/download/s3.rs

```diff
@@ -164,94 +176,75 @@ impl S3Downloader {
         tracing::debug!("Fetching from s3: {} (from {})", &key, &bucket);
 
         let source_key = &file_source.source.source_key;
-        let client = self.get_s3_client(source_key);
-        let request = client.get_object(rusoto_s3::GetObjectRequest {
-            key: key.clone(),
-            bucket: bucket.clone(),
-            ..Default::default()
-        });
+        let client = self.get_s3_client(source_key).await;
+        let request = client
+            .get_object()
+            .bucket(bucket.clone())
+            .key(key.clone())
+            .send();
 
         let source = RemoteDif::from(file_source);
         let request = tokio::time::timeout(self.connect_timeout, request);
         let request = super::measure_download_time(source.source_metric_key(), request);
 
-        let response = match request.await {
+        let response0 = request.await;
+        let response = match response0 {
             Ok(Ok(response)) => response,
-            Ok(Err(err)) => {
-                tracing::debug!("Skipping response from s3://{}/{}: {}", bucket, &key, err);
-
-                // Do note that it's possible for Amazon to return different status codes when a
-                // file is missing. 403 is returned if the `ListBucket` permission isn't available,
-                // which means that a 403 returned below may actually be for a missing file.
-                // https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html
-                match &err {
-                    RusotoError::Service(_) => return Ok(DownloadStatus::NotFound),
-                    RusotoError::Unknown(response) if response.status == StatusCode::FORBIDDEN => {
-                        return Err(DownloadError::Permissions)
+            Ok(Err(err0)) => {
+                tracing::debug!("Skipping response from s3://{}/{}: {}", bucket, &key, err0);
+                return match &err0 {
+                    ConstructionFailure(err) => {
+                        println!("ERROR: ConstructionFailure: {:?}", err);
```

Comment:
looks like you're still debugging all this.  But you could capture all in a single match arm and maybe capture it as a sentry error before throwing cancelled up the chain?

Reply:
Sounds like a good idea but I am not sure how to go about that, any pointers? Doesn't it require access to a Sentry installation and corresponding API keys?

Reply:
Looking for some guidance here as I know nothing about Sentry. SOme of these can probably grouped into a single response but the mapping from Rusoto errors to AWS SDK errors isn't clear to me (although granted I have not spent much time investigating this yet).

Also, the Rusoto code does some XML parsing of the AWS responce, I wonder how much of that is still required with the AWS SDK.